src/utils/transactions/getTransactionsHistory/helpers/mapServerTransactionsToListItems.ts

```diff
@@ -0,0 +1,91 @@
+import { getServerTransactionsByHashes } from 'apiCalls/transactions/getServerTransactionsByHashes';
+import { ITransactionListItem } from 'lib/sdkDappCoreUi';
+import { TransactionServerStatusesEnum } from 'types/enums.types';
+import { ServerTransactionType } from 'types/serverTransactions.types';
+import { cacheTransaction, getCachedTransaction } from './cacheTransactions';
+import { mapTransactionToListItem } from './mapTransactionToListItem';
+interface IMapServerTransactionsToListItemsParams {
+  hashes: string[];
+  address: string;
+  explorerAddress: string;
+  egldLabel: string;
+}
+
+export const mapServerTransactionsToListItems = async ({
+  hashes,
+  address,
+  explorerAddress,
+  egldLabel
+}: IMapServerTransactionsToListItemsParams): Promise<
+  ITransactionListItem[]
+> => {
+  const cachedTransactions: ITransactionListItem[] = [];
+  const hashesToFetch: string[] = [];
+
+  hashes.forEach((hash) => {
+    const cachedTransaction = getCachedTransaction(hash);
+    if (cachedTransaction) {
+      cachedTransactions.push(cachedTransaction);
+    } else {
+      hashesToFetch.push(hash);
+    }
+  });
+
+  if (hashesToFetch.length > 0) {
+    const newTransactions = await getServerTransactionsByHashes(hashesToFetch);
+
+    if (newTransactions.length < hashesToFetch.length) {
```

Comment:
I think we should use directly the filter below since number may give false positives